### ISSUE TITLE
old version of lmod installed on Ubuntu

### ISSUE THREAD

**Issue by djgreen** [treydock/puppet-lmod#36]
Noticed that on RHEL, our version of lmod is 8.2.7 (from 2019)
And on Ubuntu, our version of lmod is 6.6 (from 2016)

According to the docs for Ubuntu 18/20 installs (see https://lmod.readthedocs.io/en/latest/030_installing.html#using-your-package-manager-ubuntu-for-18-04-and-20-04), the required packages are newer than you're installing. 

Wondering if that's why its so far behind. 
(granted, be curious if we could get both to current 8.5)

**Comment by treydock** [comment]
I think the only thing this module can do is install different dependencies when installing lmod from source.  If you are installing Lmod via apt from OS repos, then this module has no control over the version installed.  If you want a newer Lmod on Ubuntu you will have to install from source with `install_method => source` as well as change several other parameters most likely to build Lmod to your liking.

**Comment by djgreen** [comment]
While I understand that, you will note that your defined requirements are out of date (according to their website).

**Comment by treydock** [comment]
Yep, I am going to open a pull request to bump Debian and Ubuntu to Lua 5.3 for source installs, which I think should solve that problem.  Just a little more involved than adding packages due to having to update alternatives and a symlink for lua-posix.

**Comment by treydock** [comment]
@djgreen I have #37 that will use newer Lua and tcl per Lmod docs for Debian 10 and Ubuntu 18.04 and 20.04.  If you want to test installing newer Lmod via source, give that pull request a test.

**Comment by djgreen** [comment]
Thanks, but not sure how I'd test it until you release the new version.

**Comment by treydock** [comment]
If you use r10k you could do something like this:

```
mod 'lmod', :git => 'https://github.com/treydock/puppet-lmod.git', :branch => 'debian-lua'
```

I'll go ahead and release since all the tests pass.

**Comment by treydock** [comment]
This is released as v3.1.0 once Github Actions complete

**Comment by djgreen** [comment]
OK, I deployed and nothing changed. So I went back and re-read your comments as I seem to have misunderstood them. The doc page I sent you is the instructions for how to install via packages. But your changes appear to only affect source installs.
